docs/designers-developers/developers/tutorials/javascript/js-build-setup.md

```diff
@@ -0,0 +1,228 @@
+```
+macOS: brew install node
+Windows: choco install node
+Ubuntu: apt install node
+```
+
+Additionally, the [Node.js download page](https://nodejs.org/en/download/) includes installers and binaries.
+
+**Note:** The build tools and process occur on the command-line, so some basic familiarity using a terminal application is required. Some text editors have a terminal built-in which is fine to use, Visual Studio Code editor is one popular example.
+
+### Node Package Manager
+
+The Node Package Manager (npm) is a tool included with node. NPM allows you to install and manage JavaScript packages using the package.json definition file.
+
+First, create a directory to work in.
+```
+mkdir myguten-block
+```
+
+You create a new package.json using: `npm init`.  This will walk you through creating a file.
+
+```
+$ npm init
+
+package name: (myguten-block) myguten-block
+version: (1.0.0)
+description: Test block
+entry point: (index.js) block.js
+test command:
+git repository:
+keywords:
+author: mkaz
+license: (ISC) GPL-2.0-only
+About to write to /home/mkaz/src/wp/scratch/package.json:
+
+{
+  "name": "myguten-block",
+  "version": "1.0.0",
+  "description": "Test block",
+  "main": "block.js",
+  "scripts": {
+    "test": "echo \"Error: no test specified\" && exit 1"
+  },
+  "author": "mkaz",
+  "license": "GPL-2.0-only"
+}
+
+
+Is this OK? (yes) yes
+```
+
+## Webpack & Babel
+
+The next step is to install webpack. You can install packages using the npm command: `npm install`. If you pass the `--save` parameter, npm will write the package as a dependency in package.json file.
+
+Run: `npm install --save webpack`
+
+After installing, a `node_modules` directory is created with the webpack module and its dependencies.
+
+Also, if you look at package.json file it will include a new section:
+```json
+"dependencies": {
+	"webpack": "^4.29.0"
+}
+```
+
+Next, you configure webpack to process the `block.js` file and run babel to transform the JSX.
+
+Create the file `webpack.config.js`
+
+```js
+// sets mode webpack runs under
+const NODE_ENV = process.env.NODE_ENV || 'development';
+
+module.exports = {
+	mode: NODE_ENV,
+
+	// entry is the source script
+	entry: './block.js',
+
+	// output is where to write the built file
+	output: {
+		path: __dirname,
+		filename: 'block.build.js',
+	},
+	module: {
+		// the list of rules used to process files
+		// this looks for .js files, exclude files
+		// in node_modules directory, and uses the
+		// babel-loader to process
+		rules: [
+			{
+				test: /.js$/,
+				exclude: /node_modules/,
+				loader: 'babel-loader',
+			},
+		],
+	},
+};
+```
+
+Next, you need to install babel, the webpack loader, and the JSX plugin using:
+```
+npm install --save babel-loader babel-core babel-plugin-transform-react-jsx
```

Comment:
Instead of doing this, why not use our scripts package instead. cc @gziolo. All these configs should be automatic for people. They'd only have to learn one thing.

Reply:
:+1: Will address, thanks!